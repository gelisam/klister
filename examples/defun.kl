#lang "prelude.kl"

[defun last-stx (stx)
  (syntax-case stx
    [[cons a d]
     (syntax-case d
       [() a]
       [_ (last-stx d)])]  -- note that this is a recursive call to last-stx, whereas earlier (e.g. in n-ary-app.kl)
                           -- we'd have to return the syntax of a macro call to last-stx.
    [_ stx])]

[example (last-stx '(a))]
[example (last-stx '(a b c d e f g))]
