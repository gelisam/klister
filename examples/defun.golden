#[defun.kl:12.22-12.23]<a> : Syntax
#[defun.kl:13.34-13.35]<g> : Syntax
